Sprint 1

kirafromsea opened the pull request.

Commit:
Add pages for sprint 1

mikebars1995 commented on index.html:
```diff
@@ -0,0 +1,13 @@
+    <title>Rigel chat</title>
+  </head>
+  <body>
+    <div id="app"></div>
```
**Нужно исправить**

По семантике нужно помещать весь главный контент страницы в тег `main`.


Чтобы сделать сайт более доступным для пользователя, нужно стараться придерживаться **семантической верстки**.
Несколько полезных статей на эту тему:
•	[Что такое семантическая вёрстка и зачем она нужна](https://htmlacademy.ru/blog/html/semantics)
•	[Семантика для циников](https://www.youtube.com/watch?v=B5pomLr059U)

kirafromsea replied:
Исправлено

mikebars1995 commented on src/components/Button/button.tmpl.js:
```diff
@@ -0,0 +1,3 @@
+export const buttonTmpl = `
+  <Button class="button button--{{type}}" onClick={{{onClick}}}>{{title}}</Button>
```
**Можно лучше**


Лучше давать разный `type` для разных кнопок

Например, если это кнопка сабмита, то у нее должен быть `type = "submit"`, иначе, не будет работать встроенная логика обработки формы по нажатию на `Enter`

kirafromsea replied:
Done

mikebars1995 commented on src/pages/profile/profile.tmpl.js:
```diff
@@ -0,0 +1,21 @@
+            <div><strong>E-mail:</strong> <input name="email" placeholder="Email" value={{{profile.email}}} /> </div>
+            <div><strong>Phone:</strong> <input name="phone" placeholder="Phone" value={{{profile.phone}}} /> </div>
+            {{{sendProfileButton}}}
+        </form>
```
**Нужно исправить**


На странице редактирования профиля не хватает инпута `имя в чате`, у него должен быть атрибут `name="display_name"` по заданию


По заданию 1го спринта нужно давать инпутам **конкретные** атрибуты `name`, чтобы в 3м спринте не менять ничего при подключении к `Api`. Скрин из задания https://disk.yandex.ru/i/lSPqeGrAa20ayQ

kirafromsea replied:
Done

mikebars1995 commented on server/server.js:
```diff
@@ -0,0 +1,10 @@
+import express from 'express';
+
+const app = express();
+const PORT = 4000;
```
**Можно лучше**


Лучше дать возможность поменять порт из консоли, если понадобится: `const PORT = process.env.PORT || 3000;`  . Если не задан порт, тогда будет `3000` по умолчанию.

kirafromsea replied:
Добавила здесь для сервера (4000 порт)
И в конфиге для vite (3000 порт)

mikebars1995 reviewed (commented):
Здравствуйте. (**Нужно развернуть общий комментарий ↓**)

Работа проделана огромная


Что сделано отлично:


* `Readme` хорошо оформлен
* Проект задеплоен и отлично работает
* Отлично, что не забыли про `.gitignore`
* Отлично, что ограничиваете версию `node`
* Отлично, что нет `EOF` ошибок в гите
* Хорошая структура папок и файлов
* Отлично, что цвета вынесены в `css`-константы
* Отлично, что не забываете про `alt` в тегах `img`
* Отлично, что установили язык страницы
* Отлично, что добавили `viewport`

Что нужно исправить:

* По семантике нужно помещать весь главный контент страницы в тег `main`.
* На странице редактирования профиля не хватает инпута `имя в чате`, у него должен быть атрибут `name="display_name"` по заданию



Что можно улучшить:

* Лучше давать разный `type` для разных кнопок
* Лучше дать возможность поменять порт из консоли, если понадобится: `const PORT = process.env.PORT || 3000;`  . Если не задан порт, тогда будет `3000` по умолчанию.


Исправьте, пожалуйста, недочеты и работа будет принята. Пожалуйста, проверьте работоспособность проекта и наличие возможных ошибок в консоли браузера (кнопка `F12`) перед отправкой на ревью. 

Напоминаю, что работа может быть принята только после исправления всех критических замечаний `Нужно исправить`.

**Комментарии `Можно лучше` не обязательны к исправлению прямо сейчас, это рекомендации**

Удачного рефакторинга кода.

Commit:
Fix after review
add main in index.html
add display_name in profile page
add change port in vite config
add type for buttons and inputs

mikebars1995 reviewed (commented):
Поздравляю! Ваша работа принята.

Вы отлично потрудились.

Удачного дальнейшего обучения.
